index.html

```diff
@@ -762,18 +845,148 @@ <h2><dfn>CaptureController</dfn></h2>
             [Exposed=Window, SecureContext]
             interface CaptureController {
               constructor();
-              // TODO: Add setFocusBehavior() in a separate PR.
+              undefined setFocusBehavior(CaptureStartFocusBehavior focusBehavior);
             };
           </pre>
           <dl data-link-for="CaptureController" data-dfn-for="CaptureController" class="methods">
             <dt>
               <dfn>constructor</dfn>
             </dt>
+            <dd>
+              Create a new object with the following internal slots:
+              <table class="simple">
+                <thead>
+                  <tr>
+                    <th>Internal Slot</th>
+                    <th>Initial value</th>
+                    <th>Description (<em>non-normative</em>)</th>
+                  </tr>
+                </thead>
+                <tbody>
+                  <tr>
+                    <td><dfn>[[\IsBound]]</dfn></td>
+                    <td><code>false</code></td>
+                    <td>
+                      Tracks whether an application has attempted to associate [=this=]
+                      with a [=capture-session=], by providing it as a value to
+                      {{CaptureController/setFocusBehavior()}}.
+                    </td>
+                  </tr>
+                  <tr>
+                    <td><dfn>[[\Source]]</dfn></td>
+                    <td><code>null</code></td>
+                    <td>
+                      Tracks with which source the [=capture-session=] is associated.
+                      It is distinct from {{CaptureController/[[IsBound]]}}, in that
+                      {{CaptureController/[[IsBound]]}} is set even if
+                      {{MediaDevices/getDisplayMedia()}} rejects, whereas {{CaptureController/[[Source]]}}
+                      will only be set if {{MediaDevices/getDisplayMedia()}} succeeds.
+                    </td>
+                  </tr>
+                  <tr>
+                    <td><dfn>[[\DisplaySurfaceType]]</dfn></td>
+                    <td><code>null</code></td>
+                    <td>
+                      Once capture starts, this will be set to the type of the captured [=display surface=].
+                    </td>
+                  </tr>
+                  <tr>
+                    <td><dfn>[[\InitialFocusLost]]</dfn></td>
+                    <td><code>false</code></td>
+                    <td>
+                      Ensures that calls to {{CaptureController/setFocusBehavior()}} fail
+                      without throwing if the user manually focuses another tab or native application.
```

Comment:
This raises two questions:

1. When `getDisplayMedia()` is called, the user can switch tabs but the browser prompt will still be shown until user make a decision. By monitoring focus changes **before** `getDisplayMedia` returned promise is resolved, we would allow a situation where `setFocusBehavior` would silently fail if user focused another tab while the browser prompt was displayed. The solution would be to monitor focus changes **after** `getDisplayMedia` returned promise is resolved.
2. Even with that suggested change, I think `InitialFocusLost` is not required in the spec as the "focus change" JS event would not be fired before `setFocusBehavior` be called. Am I understand this right?

What are your thoughts?

Reply:
* The question of whether the picker should persist when the user tabs away-then-back is orthogonal.
* The PR did not make it sufficiently clear that `InitialFocusLost` should only start tracking focus after the user makes their choice. That's now fixed.
* Events go on a queue. I prefer to skip it and ensure `setFocusBehavior` is rendered no-op immediately, so that a user would never find their `alt-tab` invalidated.